Sources/StreamChat/Models/ChatMessageLinkAttachment.swift

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// A type representing an attachment of type `link`.
+/// More info about enriched `URLs` can be found here:
+/// https://getstream.io/chat/docs/node/message_format/?language=swift#attachment-format
+public struct ChatMessageLinkAttachment: ChatMessageAttachment, AttachmentEnvelope, Decodable {
+    public var type: AttachmentType { .link }
+    /// A unique identifier of the attachment.
+    public var id: AttachmentId?
+    /// An original `URL` that was enriched.
+    public let ogURL: URL?
+    /// A title (e.g video name in case of enriched `YouTube` link or song name in case of `Spotify` link).
+    public let title: String?
+    /// A text, usually description of the link content.
+    public let text: String?
+    /// An author, usually the link origin. (e.g. `YouTube`, `Spotify`)
+    public let author: String?
+    /// A link for displaying an attachment.
+    /// Can be different from the original link, depends on the enriching rules.
+    public let titleLink: URL?
+    /// An image.
+    public let imageURL: URL?
+    /// A preview image.
+    public let thumbURL: URL?
+    /// A `URL` for an asset this link points to.
+    public let assetURL: URL?
+        
+    private enum CodingKeys: String, CodingKey {
+        case type
+        case title
+        case text
+        case author = "author_name"
+        case ogURL = "og_scrape_url"
+        case titleLink = "title_link"
+        case imageURL = "image_url"
+        case thumbURL = "thumb_url"
+        case assetURL = "asset_url"
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
```

Comment:
We should also check that the type matches the expected one. It's possible that there will be a custom attachment with the same structure but different type.

Reply:
Nice, thanks!